### PR TITLE
Run unit tests on Windows too

### PR DESCRIPTION
**Pull request by henrymercer**
This PR runs our unit tests on Windows, primarily to improve test coverage but also to support developing the Action on Windows machines.  In doing so, we fix a couple of bugs.

For review: I spent a while looking at whether we could avoid the ugly hack in https://github.com/github/codeql-action/pull/1099/commits/6063828d4d44ed49b31138f425c444c389c684cd, however couldn't find anything.  Do you have any thoughts?

Based on #1104.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Review by aeisenberg — COMMENTED** [review]
This looks sensible. I just have a few questions inline so I can understand what is going on.

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -90,14 +90,16 @@ jobs:
...
+        os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 45
 
     steps:
       - uses: actions/checkout@v3
       - name: npm run-script test
-        run: npm run-script test
+        run: |
+          npm install
```
[the remark this answers is not in the document]
Comment (henrymercer) [reply .github/workflows/pr-checks.yml:101]: The problem was that Windows couldn't find things that lived in `node_modules/.bin/` like `ava`.  I found a better solution though, which is to install `npm config set script-shell bash`.  Have implemented.

File: src/util.ts [hunk 2]
```diff
@@ -112,11 +112,7 @@ export async function withTmpDir<T>(
   body: (tmpDir: string) => Promise<T>
 ): Promise<T> {
   const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), "codeql-action-"));
-  const realSubdir = path.join(tmpDir, "real");
-  fs.mkdirSync(realSubdir);
-  const symlinkSubdir = path.join(tmpDir, "symlink");
-  fs.symlinkSync(realSubdir, symlinkSubdir, "dir");
-  const result = await body(symlinkSubdir);
+  const result = await body(tmpDir);
```
[the remark this answers is not in the document]
Comment (henrymercer) [reply src/util.ts:115]: This is documented in the commit message https://github.com/github/codeql-action/pull/1099/commits/bdf7aad654905cf5635cc54da906a25ffb938974:

> `toolcache.extractTar` currently falls over when `ACTIONS_TEMP` contains a symlink, and the runner no longer exists, so it's unlikely our customers would be running with temporary directories that contain symlinks.

Let me know if you'd like additional clarification on anything.

[the remark this answers is not in the document]
Comment (aeisenberg) [reply src/util.ts:115]: That's fine. I didn't look at the individual commits. Thanks for clarifying.

**Review by aeisenberg — APPROVED** [review]

**Review by aeisenberg — APPROVED** [review]

**Pull request merged**